### DISCUSSION TITLE
File configuration!

### DISCUSSION THREAD

**Discussion by SrMiaguii** [make-all/tuya-local#2024] [Q&A] [answered]
Hello everybody! Where do I find the configuration file for the added devices. I'm wondering which device*.yaml file I used and which protocol I used for certain devices. And once found, could I edit them?

**Comment by make-all** [comment, marked as answer]
You can download the device diagnostics to check which device yaml file you used.  The configuration file is not intended to be edited by hand, any parameters that are safe to edit are available for editing in the device Options.  The device id is used as a unique id, so to change that you would need to remove and re-add the device.  The yaml file affects what entities are available, so also requires removal and re-adding to ensure the old entities are cleaned up.

**Reply by SrMiaguii** [reply]
Understood! Thank you very much again for your kindness!
